src/hooks/useRecentSearches.ts

```diff
@@ -1,41 +1,45 @@
-import { useCallback, useEffect, useRef, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import RecentSearches, { ISearch } from 'recent-searches';
 
-export const RECENT_SEARCHES_KEY = '__yxt_recent_searches__';
-
 export function useRecentSearches(
-  recentSearchesLimit: number
-): [ISearch[]|undefined, (input: string) => void, () => void] {
-  const recentSearchesLimitRef = useRef(recentSearchesLimit);
-  const [ recentSearches, setRecentSeaches ] = useState<RecentSearches>(
+  recentSearchesLimit: number,
+  verticalKey: string | null
```

Comment:
It might be sliightly simpler to use undefined instead of null here? I like that null is more "clear" than undefined, but accepting undefined here instead let's us get rid of defaulting verticalKey to null above in `SearchBar`(currently line 190)

Reply:
I see, that looks better